process/factory/shard/interceptorsContainerFactory.go

```diff
@@ -211,21 +236,28 @@ func (icf *interceptorsContainerFactory) generateTxInterceptors() ([]string, []p
 }
 
 func (icf *interceptorsContainerFactory) createOneTxInterceptor(identifier string) (process.Interceptor, error) {
-	//TODO implement other TxHandlerProcessValidator that will check the tx nonce against account's nonce
-	txValidator, err := dataValidators.NewNilTxValidator()
+	argProcessor := &processor.ArgTxInterceptorProcessor{
+		ShardedDataCache: icf.dataPool.Transactions(),
+	}
+	txProcessor, err := processor.NewTxInterceptorProcessor(argProcessor)
 	if err != nil {
 		return nil, err
 	}
 
-	interceptor, err := transaction.NewTxInterceptor(
+	txFactory, err := interceptorFactory.NewInterceptedDataFactory(
+		icf.argInterceptorFactory,
```

Comment:
Partially agree because: 1) the parameter count will be higher, 2) will end up depending on 3 or 4 large structs that will hold just about anything in our project. (will be 3 or 4 sub-God clases). 3) test files will have a large dedicated area just to instantiate those large components structs instead of having just a smaller section for arg instantiation.